mypy/types.py

```diff
@@ -1162,23 +1162,45 @@ class TypeType(Type):
     # a generic class instance, a union, Any, a type variable...
     item = None  # type: Type
 
+    # this class should not be created directly
+    # use make method which may return either TypeType or UnionType
+    # this is to ensure Type[Union[A, B]] is always represented as Union[Type[A], Type[B]]
     def __init__(self, item: Type, *, line: int = -1, column: int = -1) -> None:
```

Comment:
The `__init__` argument type wouldn't accept `UnionType` so mypy can prevent you from calling it with the wrong argument type. Calling `TypeType(item)` would be disallowed if `item` has type `Type`. 

The signature of `__init__` could be something like this (not sure about the exact union type):

```py
def __init__(self, item: Union[AnyType, Instance], ...) -> None:   # no Type allowed!
    ...
```

You can also mention the alternative way of constructing `TypeType` instances in the docstring for `__init__`.